p2p/src/peers/mod.rs

```diff
@@ -201,7 +320,10 @@ impl Peers {
                 {
                     log::trace!("Removed a tried peer address: \n{}", self);
 
-                    self.tried_bucket.remove(&bucket_index)
+                    self.tried_bucket.remove(&bucket_index).map(|entry| {
+                        self.ice_peer_address(&entry.address);
```

Comment:
In the try_peer method (https://github.com/witnet/witnet-rust/blob/master/node/src/actors/peers_manager/mod.rs#L106), there is a hack where the tried peer is removed, but is not wrong.

In this case, when a new peer wants to use a tried index that is already in use, but it is not tested from a long time, first of all, that peer is removed from tried, and before to try the new peer, the old one is tried. Currently the index is empty, so if the old one success in the handshake, it will be added again to the tried bucket but with a updated timestamp, if not, the index will be free for the next filler try adding a new peer in that index

Reply:
:trollface: